src/machine/usb.go

```diff
@@ -72,6 +72,22 @@ var udd_ep_in_cache_buffer [7][64]uint8
 //go:align 4
 var udd_ep_out_cache_buffer [7][64]uint8
 
+var usb_trans_buffer []uint8 = makeUSBTransBuffer()
```

Comment:
You could probably also pre-allocate this as `var usb_trans_buffer [whatever the max size]uint8` then `return usb_trans_buffer[:size]` in the following function.

Reply:
Looking further into what you did, I think you already probably have enough for now.

Reply:
I've pushed a version that pre-allocates at max size.  The trade-off is probably something like costing 200 bytes, but is a bit simpler.